Source/Atomic/Resource/Image.cpp

```diff
@@ -2034,7 +2124,12 @@ unsigned char* Image::GetImageData(Deserializer& source, int& width, int& height
 
     SharedArrayPtr<unsigned char> buffer(new unsigned char[dataSize]);
     source.Read(buffer.Get(), dataSize);
-    return stbi_load_from_memory(buffer.Get(), dataSize, &width, &height, (int*)&components, 0);
+    // #623 BEGIN TODO: Crunch expects all 4 components. This isn't ideal, but since
+    // precompressed versions should always be loaded anyway it doesn't matter.
+    // Might need to handle differently per platform.
+    components = 4;
+    return stbi_load_from_memory(buffer.Get(), dataSize, &width, &height, nullptr, components);
+    // #623 END TODO: Crunch expects all 4 components. This isn't ideal, but since

```

Comment:
With SaveDDS being suppoorted on all desktop platforms, that does mean we'd need to do this on all of those as well, just a guard change, but just FYI.

Resulting code in BeginLoad would be:
`
#if defined(ATOMIC_PLATFORM_WINDOWS) || defined (ATOMIC_PLATFORM_LINUX) || defined (ATOMIC_PLATFORM_OSX)
        // #623 BEGIN TODO: Crunch expects all 4 components. This isn't ideal, but since
        // precompressed versions should always be loaded anyway it doesn't matter.
        // Might need to handle differently per platform.
        // #623 END TODO: Crunch expects all 4 components. This isn't ideal, but since
        unsigned char* pixelData = GetImageDataForceAlphaLayer(source, width, height, components);
#else
        unsigned char* pixelData = GetImageData(source, width, height, components);
`

Still including components as a reference param is a tossup, but it keeps this cleaner.